sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/AllBranchesShouldNotHaveSameImplementation.cs

```diff
@@ -144,5 +144,17 @@ private void DoSomething()
         private void DoSomethingElse()
         {
         }
+
+        public int Test(string type)
+        {
+            return type switch // Compliant - FN: all branches call the same function
+            {
+                "a" => GetNumber(),
```

Comment:
would it be useful to add
- a test with nested identical switch expressions?
- compliant tests for switch expressions